src/Mono.Android/Android.Runtime/JavaProxyThrowable.cs

```diff
@@ -1,23 +1,84 @@
 using System;
+using System.Diagnostics;
+using System.Reflection;
+
+using StackTraceElement = Java.Lang.StackTraceElement;
 
 namespace Android.Runtime {
 
-	class JavaProxyThrowable : Java.Lang.Error {
+	sealed class JavaProxyThrowable : Java.Lang.Error {
 
 		public  readonly Exception InnerException;
 
-		public JavaProxyThrowable (Exception innerException)
-			: base (GetDetailMessage (innerException))
+		JavaProxyThrowable (string message, Exception innerException)
+			: base (message)
 		{
 			InnerException  = innerException;
 		}
 
-		static string GetDetailMessage (Exception innerException)
+		public static JavaProxyThrowable Create (Exception? innerException, bool appendJavaStackTrace = false)
+		{
+			if (innerException == null) {
```

Comment:
Why have `Exception? innerException` if you're gonna throw `ArgumentNullException` here?

Reply:
It was a typo